v0/destinations/refiner/utils.js

```diff
@@ -0,0 +1,210 @@
+ * @param {*} destination
+ * @returns
+ */
+const getAccountTraits = (message, destination) => {
+  const { accountAttributesMapping } = destination.Config;
+  const traits = getFieldValueFromMessage(message, "groupTraits");
+  const id = getFieldValueFromMessage(message, "groupId");
+  let groupUsersPayload = { ...traits, id };
+  const accountAttributesMap = getHashFromArray(
+    accountAttributesMapping,
+    "from",
+    "to",
+    false
+  );
+  groupUsersPayload = replaceDestAttributes(
+    accountAttributesMap,
+    groupUsersPayload
+  );
+  return groupUsersPayload;
+};
+
+/**
+ * Returns identify event payload
+ * ref : https://refiner.io/docs/api/#identify-user
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const identifyUserPayloadBuilder = (message, destination) => {
+  const { userAttributesMapping } = destination.Config;
+  const userId = getFieldValueFromMessage(message, "userIdOnly");
+  const email = getFieldValueFromMessage(message, "email");
+  const traits = getFieldValueFromMessage(message, "traits");
+  const contextTraits = get(message, "context.traits");
+  let payload = { ...traits, ...contextTraits, userId, email };
+  const userAttributesMap = getHashFromArray(
+    userAttributesMapping,
+    "from",
+    "to",
+    false
+  );
+  payload = replaceDestAttributes(userAttributesMap, payload);
+  const { endpoint } = CONFIG_CATEGORIES.IDENTIFY_USER;
+  return { payload, endpoint };
+};
+
+/**
+ * Returns track event payload
+ * ref : https://refiner.io/docs/api/#track-event
+ * @param {*} message
+ * @returns
+ */
+const trackEventPayloadBuilder = message => {
+  const payload = {};
+  const userId = getFieldValueFromMessage(message, "userIdOnly");
+  const email = getFieldValueFromMessage(message, "email");
+  const { event } = message;
+  if (userId) {
+    payload.id = userId;
+  }
+  if (email) {
+    payload.email = email;
+  }
+  payload.event = event;
+  const { endpoint } = CONFIG_CATEGORIES.TRACK_EVENT;
+  return { payload, endpoint };
+};
+
+/**
+ * Returns group event payload
+ * ref : https://refiner.io/docs/api/#group-users
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const groupUsersPayloadBuilder = (message, destination) => {
+  const payload = {};
+  const userId = getFieldValueFromMessage(message, "userIdOnly");
+  const email = getFieldValueFromMessage(message, "email");
+  if (userId) {
+    payload.id = userId;
+  }
+  if (email) {
+    payload.email = email;
+  }
+  payload.account = getAccountTraits(message, destination);
+  const { endpoint } = CONFIG_CATEGORIES.GROUP_USERS;
+  return { payload, endpoint };
+};
+
+/**
+ * Handles the destination Response and return it to rudder-server
+ * @param {*} destinationResponse
+ * @returns
+ */
+const responseHandler = destinationResponse => {
+  const message = `[Refiner] - Request Processed Successfully`;
+  const { status } = destinationResponse;
+  if (isHttpStatusSuccess(status)) {
```

Comment:
and if it is not successful, how we are handling that? Please take a reference from Appsflyer PR and handle that

Reply:
@shrouti1507 you can see the line no 182 to 190 will be executed if event is not successful, we are already handling it